docs/en/03_Upgrading/index.md

```diff
@@ -37,123 +89,1015 @@ and also checking the `composer.json` files in [recipe-core](https://github.com/
+    SINGULARNAME: Coyote
+  Acme\Web\RoadRunner:
+    SALUTATION: Beep Beep
+```
+
+### Finalising namespace updates
+Whatever you upgraded your code automatically or manually you'll need to perform the two following steps manually.
+
+DataObject tables will default to use a namespaced name. For example, if you have class under `Acme\Web\Products\ExplosiveTennisBall` that extends `DataObject`, the matching table in your database will be called `Acme_Web_Products_ExplosiveTennisBall`.
+
+You can define a `private static $table_name` property on your DataObjects to use more convenient table names. For example, `private static $table_name = 'ExplosiveTennisBall';`.
+
+If your template files contain translatable strings, they also need to be updated to referenced the namespaced classes.
+
+For example, `<%t Member.SINGULARNAME 'Member' %>` would become `<%t SilverStripe\Security\Member.SINGULARNAME 'Member' %>`.
+
+Your codebase should now be referencing valid SilverStripe 4 classes. This means that your classes can be loaded at runtime. However, your codebase will still be using an outdated API.
+
+This is a good point to commit your changes to your source control system before moving on to the next step.
+
+## Step 5 - Updating your codebase to use SilverStripe 4 API {#step5}
+
+This is the most intricate and potentially time-consuming part of the upgrade. Basically, it involves going through your entire codebase to remove references to deprecated APIs and update your project logic.
+
+### Automatically update deprecated API references with the `inspect` command
+
+The upgrader has an `inspect` command that can flag deprecated API usage, and in some cases, update your codebase to the SilverStripe 4 equivalent. This does require you to carefully review each change and warning to make sure the updated logic still work as intended. Even so, it is a huge time-saver compared to reviewing your code base manually.
+
+Note that the `inspect` command loads your files with PHP interpreter. So basic syntax errors — for example, extending a class that does not exists — will cause an immediate failure. For this reason, you need to complete [Step 4 - Update codebase with references to newly namespaced classes](#step4) before running the `inspect` command.
+
+```bash
+/var/www/SS_acme > upgrade-code inspect ./mysite/ --write
+```
+
+You can omit the `--write` flag if you just want to view the proposed changes without applying them. You can run the command on a specific subdirectory or file. This can be more manageable if you have a big project to go through.
+
+Like the `upgrade` command, `inspect` gets its list of API changes from `.upgrade.yml` files. So you may get upgrade suggestions and substitution from third party modules. You can even include your own project specific changes in your `.upgrade.yml` if you want.
+
+#### Sample output of the `inspect` command
+Here's some sample output of what you might get back the `inspect` command.
+
+```bash
```

Comment:
Good idea to provide sample output!

Reply:
I think it's the only place I provide an exact console output. Most other places, I did a before-and-after.